Implement drag and drop feature for the FileList

BohdanDudkaTD opened the pull request:
This PR implements logic to drag and drop files from the OS of the user to the list of files.
PR adds `StorageFile` - model of the file which is stored on the server. It might be a simple file or folder with files.
Also, it adds `FileListItem` - component which represents list item of the `FileList` component. Each `FileListItem` has its handlers for the drag and drop logic.
Besides, that handlers of drag and drop logic were added to the `FileList` component.

Result of the ESLint execution:
![image](https://user-images.githubusercontent.com/50625022/64286058-6edaf280-cf65-11e9-8e06-bee43fd971d8.png)

BohdanDudkaTD commented:
@dkalinin, PTAL

dkalinin commented on app/store-configuration/action/index.js:
```diff
@@ -9,8 +9,20 @@ import {
-  [ADD_ITEM_ACTION](context, payload) {
-    context.commit(ADD_ITEM_MUTATION, payload);
+  [ADD_ITEM_ACTION](context, currentFolderId, folderId, files) {
+    context.commit(CHANGE_ARE_FILES_LOADING_MUTATION, true);
```
SET_FILES_LOADING

dkalinin commented on app/view/component/file-list-item/index.js:
```diff
@@ -0,0 +1,160 @@
+   *
+   * @param {HTMLElement} parent -  element in which this component should be rendered
+   * (it might be an ol or an ul).
+   * @param {StorageFile} item - the instance of file model.
```
item => file

dkalinin commented on app/view/component/file-list/index.js:
```diff
@@ -33,25 +37,113 @@ export class FileList extends ConfigurableComponent {
         });
 
-        this._fileList.appendChild(listItem);
+        if (item.type === 'folder') {
```
If item.type is file nothing happens so there's no need to add event listeners in FileItem component.

dkalinin commented on app/view/component/file-list-item/index.js:
```diff
@@ -0,0 +1,160 @@
+   */
+  removeClass(className) {
+    this._dragCounter--;
+    if (this._dragCounter === 0) {
```
Adding or removing classes is not related to drag events. Please split this logic.

BohdanDudkaTD commented:
@dkalinin, PTAL, fixed issues:
- renamed variables;
- logic to set/remove classes of 
HTML element was separated from the drag events logic;
- `DragAndDrop` component was added as a parent class for all components, which supports drag and drop events. Now each component sets event listeners only when it needs it;

dkalinin commented on app/model/storage-file/index.js:
```diff
@@ -19,58 +19,4 @@ export class StorageFile {
     this._type = type;
     this._fileType = fileType;
```
Fields should be public.

dkalinin commented on app/view/component/file-list/index.js:
```diff
@@ -86,63 +64,26 @@ export class FileList extends ConfigurableComponent {
     event.stopPropagation();
-    this._dragCounter--;
+    this.onDragLeave(event, this._fileList);
     this._fileList.classList.remove('dragover');
```
Please remove this line.
